app/models/hyrax/pcdm_file_set.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Valkyrie model for `FileSet` domain objects in the Hydra Works model.
+  #
+  # @see https://wiki.duraspace.org/display/samvera/Hydra%3A%3AWorks+Shared+Modeling
+  class PcdmFileSet < Hyrax::Resource
```

Comment:
Did we decide `FileSet` wouldn't work?

Reply:
Yes, it doesn't work because Hyrax::FileSet is already defined as a module.

As I stated in the PR comments, I am open to a different name.  FileSets are not part of the PCDM model.  They are part of the Works model.  But it seems very confusing to call this WorksFileSet.  So I went with the slightly incorrect PcdmFileSet (and also potentially confusing)

Perhaps a better solution that would work for all the base resources would be to name them something like...

```
Hyrax::WorksModel::Collection
Hyrax::WorksModel::Work
Hyrax::WorksModel::FileSet
```

Reply:
>  Hyrax::FileSet is already defined as a module.

The current module is just a namespace module, right? Shouldn't it be possible to redefine it as a class, and continue to nest the existing modules, etc... in it?

----

Namespacing is an option.

My preference, though, is to just name everything `Hyrax::[ModelName]`. Along these lines, I consider `PcdmCollection` a stopgap measure until a rename in Hyrax 4.0.

Reply:
I'm not sure I am following the proposed refactor of Hyrax::FileSet module to Hyrax::FileSet class with nested modules under it?  Can you provide an example?

Reply:
See #4086.

Reply:
DONE - rebased on the Hyrax::FileSet class defined in #4086